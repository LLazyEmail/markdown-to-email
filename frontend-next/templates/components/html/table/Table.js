import React from 'react';

const Table = ({className, children, style, width, align}) => (
  <table
    align={align}
    className={className}
    border={0}
    cellPadding={0}
    cellSpacing={0}
    style={style}
    width={width}
  >
    {children}
  </table>
);

export default Table;
