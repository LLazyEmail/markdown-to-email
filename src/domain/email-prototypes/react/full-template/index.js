import { generateReactFullTemplate } from '../../../../parserMDReact/methods';

// import {
//   generateReactContent,
// } from './parserMDReact';


export {
    generateReactFullTemplate
}