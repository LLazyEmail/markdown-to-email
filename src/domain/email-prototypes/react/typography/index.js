import headingLiteral from './literals/heading';
import imageLiteral from './literals/image';
import italicLiteral from './literals/italic';
import linkLiteral from './literals/link';
import listLiteral from './literals/list';
import listItemLiteral from './literals/listItem';
import titleLiteral from './literals/mainTitle';
import paragraphLiteral from './literals/paragraph';
import qLiteral from './literals/q';
import strongLiteral from './literals/strong';
import subtitleLiteral from './literals/subtitle';
import separatorLiteral from './literals/separator';
import blockquoteLiteral from './literals/blockquote';
import codeLiteral from './literals/code';
import delLiteral from './literals/del';

export {
  headingLiteral,
  imageLiteral,
  italicLiteral,
  linkLiteral,
  listLiteral,
  listItemLiteral,
  titleLiteral,
  paragraphLiteral,
  qLiteral,
  strongLiteral,
  subtitleLiteral,
  separatorLiteral,
//----------
  blockquoteLiteral,
  codeLiteral,
  delLiteral
};
