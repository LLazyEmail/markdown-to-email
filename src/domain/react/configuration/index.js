import {
    replaceMarkdown,
} from '../replace-markdown/replace-md';


// import Replacer from '../../parserMDReact/replace';

// function configureReplacer(state, isFull) {
//   // CAN WE MOVE OUT CALLBACKS INTO REPLACER?????
//   ReactReplacer.replaceMDBinded = replaceMarkdown.bind(state);

//   // I think we didnt use it quite often
//   // ReactReplacer.replaceMDBindedPreviewText = replaceMarkdownPreviewText.bind(state);

//   // NOTE DON'T CHANGE ORDER OF FUNCTION CALLS
//   if (isFull) {
//     ReactReplacer.previewText();
//   }

//   ReactReplacer.comments();
//   ReactReplacer.typography();

//   ReactReplacer.ulList();
//   // TODO UPDATE OLLIST CALLBACK, IT'S NOT WORKING
//   //   ReactReplacer.olList()

//   ReactReplacer.blockQuote();
//   ReactReplacer.hr();
//   ReactReplacer.paragraph();

//   // TODO CHECK CASES WHEN CONTENT CAN HAVE EMPTY LISTS OR BLOCKQUOTE
//   //   ReactReplacer.emptyUl();
//   //   ReactReplacer.emptyOl();
//   //   ReactReplacer.emptyBlockQuote();

//   ReactReplacer.br();
//   ReactReplacer.sponsorship();
//   ReactReplacer.mem();
//   ReactReplacer.separator();
// }

// export default configureReplacer;
