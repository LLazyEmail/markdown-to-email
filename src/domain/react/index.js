import mainObject from './pre-replace-object';
import configureReplacer from './configuration';
export { mainObject, configureReplacer };