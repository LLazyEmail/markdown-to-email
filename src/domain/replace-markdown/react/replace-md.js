// import debuggingReplacer from './debugging';
// import { replaceMarkdown } from '../../replace-class';

const prepOurCallback = (name, configurationMap) => {
  if (!name) {
    throw new Error(`name of ${name} is undefined or empty`);
  }

  const settings = configurationMap[name];

  if (!settings) {
    throw new Error(`object by name ${name} not found`);
  }

  return settings;
};

// if (debug) {
//   //--------------
//   if (!settings.literal) {
//     console.log(callbackName, settings.constant);
//     console.log('-----------------');
//   }
//   //---------------
// }
// TODO we finally can replaceMarkdown()*** for this one.

export function replaceMarkdownReact(callback_name) {
  // const debugging = new replaceMarkdownDebug(callback_name);

  const singleElement = prepOurCallback(callback_name, this.configurationMap);

  // --------- comment for debugging purposes
  // // --- You can comment this when you debugging our wrapper
  // const _replacer = singleElement.replacer.bind(this);
  // this.content = this.content.replace(singleElement.constant, _replacer);

  const _replacer = singleElement.replacer.bind(this);
  this.content = this.content.replace(singleElement.constant, _replacer);

  //   // -------------- Uncomment for debugging reasons
  // if (debuggingReplacer(callback_name)) {
  //   // console.log(callback_name);

  //   //     console.log(replacedString);

  //   //     // TODO another case why i dont like this solution
  //   //     // with string.replace - when you forget to return something it's hard to catch
  // }
  // return '';
  //-----------------------
}

// function replaceMarkdownDebug(callback_name) {
//   // -------------- Uncomment for debugging reasons
//   if (debuggingReplacer(callback_name)) {

//     console.log(callback_name);

//     const _replacer = singleElement.replacer.bind(this);

//     console.log(replacedString);

//     // TODO another case why i dont like this solution
//     // with string.replace - when you forget to return something it's hard to catch
//     this.content = this.content.replace(singleElement.constant, _replacer);
//   }
//   return '';
//   //--------------
// }

// TODO mayber replace it with class?
// class replaceMarkdownReact {
//   constructor(callback_name) {
//     const singleElement = prepOurCallback(callback_name, false);
//     // --------- comment for debugging purposes
//     // // --- You can comment this when you debugging our wrapper
//     // const _replacer = singleElement.replacer.bind(this);
//     // this.content = this.content.replace(singleElement.constant, _replacer);
//     //   // -------------- Uncomment for debugging reasons
//     if (debuggingReplacer(callback_name)) {

//       // console.log(callback_name);
//       const _replacer = singleElement.replacer.bind(this);

//       //     console.log(replacedString);
//       //     // TODO another case why i dont like this solution
//       //     // with string.replace - when you forget to return something it's hard to catch
//       this.content = this.content.replace(singleElement.constant, _replacer);

//     }
//     return '';
//     //   //-----------------------
//   }
// }
