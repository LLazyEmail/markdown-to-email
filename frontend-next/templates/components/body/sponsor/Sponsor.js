import React from 'react';
import Table from '../../html/table';
import { styles } from './styles';

const Sponsor = ({
  leftHref, leftSrc, rightHref, children,
}) => (
  <Table
    className="mcnImageCardRightContentOuter"
    width="100%"
    style={styles.imageCardRightContentOuter}
  >
    <tbody>
      <tr>
        <td align="center" valign="top" className="mcnImageCardRightContentInner" style={styles.imageCardRightContentInner}>
          <Table
            align="left"
            className="mcnImageCardRightImageContentContainer"
            width={200}
            style={styles.imageCardRightContentContainer}
          >
            <tbody>
              <tr>
                <td className="mcnImageCardRightImageContent" align="center" valign="top" style={styles.imageCardRightContent}>
                  <a href={leftHref} data-testid="sponsorHrefTest" target="_blank" style={styles.title}>
                    <img alt="" src={leftSrc} width={150} data-testid="sponsorSrcTest" style={styles.image} className="mcnImage" />
                  </a>
                </td>
              </tr>
            </tbody>
          </Table>
          <Table
            className="mcnImageCardRightTextContentContainer"
            align="right"
            width={346}
            style={styles.imageCardRightContentContainer}
          >
            <tbody>
              <tr>
                <td valign="top" className="mcnTextContent" style={styles.textContent}>
                  <h1 className="mc-toc-title" style={styles.docTitle}>
                    <br />
                    <a href={rightHref} style={styles.link}>
                      <strong data-testid="sponsorChildrenTest">
                        {children}
                      </strong>
                    </a>
                  </h1>
                </td>
              </tr>
            </tbody>
          </Table>
        </td>
      </tr>
    </tbody>
  </Table>
);

export default Sponsor;
