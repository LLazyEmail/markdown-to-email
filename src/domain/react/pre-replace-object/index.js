import typography from 'atherdon-newsletter-react-layouts-typography';
import { Callbacks } from '../../../callbacksReact';

import {
  REGEXP_HEADER,
  REGEXP_IMAGE,
  REGEXP_LINK,
  REGEXP_STRONG,
  REGEXP_DEL,
  REGEXP_Q,
  REGEXP_CODE,
  // REGEXP_UL_LIST,
  // REGEXP_OL_LIST,
  REGEXP_BLOCKQUOTE,
  REGEXP_HR,
  REGEXP_BR,
  REGEXP_EM,
  // REGEXP_SPONSORSHIP,
  // REGEXP_MEM,
  // REGEXP_PREVIEW_TEXT,
  REGEXP_PARAGRAPH,
  // REGEXP_SEPARATOR,
} from '../../../constants/index';

import objectBuilder from '../../md/object-builder';

const {
  _strong,
  _link,
  _blockquote,
  //   _mem,
  _header,
  _italic,
  _del,
  _q,
  _code,
  _hr,
  //   _empty,
  //------------
  // _ulList,
  // _olList,
  //------------
  _image,
  _paragraph,
  // _sponsorship,
  _br,
  //------------
  // _separator,

  // _previewText,
} = Callbacks;

const {
  headingComponent,
  imageComponent,
  italicComponent,
  linkComponent,
  mainTitleComponent,
  paragraphComponent,
  strongComponent,
  subtitleComponent,
  qComponent,
  codeComponent,
} = typography;

const map = {
  strong: objectBuilder(REGEXP_STRONG, _strong, strongComponent),
  link: objectBuilder(REGEXP_LINK, _link, linkComponent),
  del: objectBuilder(REGEXP_DEL, _del, false),
  image: objectBuilder(REGEXP_IMAGE, _image, imageComponent),
  // TODO find out what module to use to import previewText
  // previewText: objectBuilder(
  //   REGEXP_PREVIEW_TEXT,
  //   _previewText,
  //   previewTextLiteral,
  // ),
  italic: objectBuilder(REGEXP_EM, _italic, italicComponent),
  header: objectBuilder(REGEXP_HEADER, _header, headingComponent),
  // TODO header arent working as suppose too
  subtitle: objectBuilder(REGEXP_HEADER, _header, subtitleComponent),
  title: objectBuilder(REGEXP_HEADER, _header, mainTitleComponent),
  q: objectBuilder(REGEXP_Q, _q, qComponent),
  code: objectBuilder(REGEXP_CODE, _code, codeComponent),

  // listItem:objectBuilder(
  //   REGEXP_SUB_LISTS,
  //   getParsedSubList,
  //   listItemLiteral
  //   ),

  // this object used only as a stupid way to add a parch for different cases of lists
  // first two params never used
  // list:objectBuilder(REGEXP_UL_LIST, getParsedSubList, listLiteral),
  // ulList:objectBuilder(REGEXP_UL_LIST, _ulList, listLiteral),
  // olList:objectBuilder(REGEXP_OL_LIST, _olList, false),
  blockquote: objectBuilder(REGEXP_BLOCKQUOTE, _blockquote, false),
  hr: objectBuilder(REGEXP_HR, _hr, false),
  paragraph: objectBuilder(REGEXP_PARAGRAPH, _paragraph, paragraphComponent),
  br: objectBuilder(REGEXP_BR, _br),
};
// sponsorship:objectBuilder(
//   REGEXP_SPONSORSHIP,
//   _sponsorship,
//   sponsorLiteral,
// );

// // //  memeLiteral,
// memes:objectBuilder(REGEXP_MEM, _meme, false);

// separator:objectBuilder(
//   REGEXP_SEPARATOR,
//   _separator,
//   separatorLiteral
//   );

// // 'REGEXP_EMPTY_UL': {
// //   constant: REGEXP_EMPTY_UL,
// //   replacer: emptyUl
// // },

// // 'REGEXP_EMPTY_OL': {
// //   constant: REGEXP_EMPTY_OL,
// //   replacer: emptyOl
// // },

// // 'REGEXP_EMPTY_BLOCKQUOTE': {
// //   constant: REGEXP_EMPTY_BLOCKQUOTE,
// //   replacer: emptyBlockQuote
// // },

export default map;
