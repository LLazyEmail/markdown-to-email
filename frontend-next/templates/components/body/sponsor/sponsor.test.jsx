import React from 'react';
import { create } from 'react-test-renderer';
import ReactDOM from 'react-dom';
import { cleanup, render } from '@testing-library/react';
import Table from '../../html/table';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Sponsor from './index';

configure({ adapter: new Adapter() });


afterEach(() => {
  cleanup();
});

describe('sponsor/index component', () => {

    


    test('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Sponsor />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    test('renders correctly', () => {
        const tree = create(<Sponsor />);
        expect(tree.toJSON()).toMatchSnapshot();
    });

    test('compoents props contains anything', () => {
        const component = create(<Sponsor><h1>Hello</h1></Sponsor>);
        const { root } = component;

        expect(root.props.children).not.toBeNull();
    });

    test('strong contains correct children elements', () => {
        const { getByTestId } = render(<Sponsor><h1>Hello</h1></Sponsor>);
        const strong = getByTestId('sponsorChildrenTest');

        expect(strong.children[0].textContent).toBe('Hello');
    });

    test('compoents children element should contain correct element', () => {
        const component = create(<Sponsor><h1>Hello</h1></Sponsor>);
        const { root } = component;

        expect(root.props.children.type).toBe('h1');
    });

    test('compoents props elements should contain correct data', () => {
        const component = create(<Sponsor leftHref="https://www.youtube.com" leftSrc="../main/img.png" rightHref="https://vk.com/" />);
        const { root } = component;

        expect(root.props.leftHref).toBe('https://www.youtube.com');
        expect(root.props.rightHref).toBe('https://vk.com/');
        expect(root.props.leftSrc).toBe('../main/img.png');
    });

    test('<a> has correct href', () => {
        const { getByTestId } = render(<Sponsor leftHref="https://www.youtube.com/" leftSrc="../main/img.png" rightHref="https://vk.com/" />);
        const a = getByTestId('sponsorHrefTest');

        expect(a.href).toBe('https://www.youtube.com/');
    });

    test('<img> has correct src', () => {
        const { getByTestId } = render(<Sponsor leftHref="https://www.youtube.com" leftSrc="main/img.png" rightHref="https://vk.com/" />);
        const img = getByTestId('sponsorSrcTest');

        expect(img.src).toBe('http://localhost/main/img.png');
    });
});
