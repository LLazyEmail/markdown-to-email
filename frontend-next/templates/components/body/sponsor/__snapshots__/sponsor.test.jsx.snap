// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sponsor/index component renders correctly 1`] = `
<table
  border={0}
  cellPadding={0}
  cellSpacing={0}
  className="mcnImageCardRightContentOuter"
  style={
    Object {
      "WebkitTextSizeAdjust": "100%",
      "borderCollapse": "collapse",
      "msTextSizeAdjust": "100%",
      "msoTableLspace": "0pt",
      "msoTableRspace": "0pt",
    }
  }
  width="100%"
>
  <tbody>
    <tr>
      <td
        align="center"
        className="mcnImageCardRightContentInner"
        style={
          Object {
            "WebkitTextSizeAdjust": "100%",
            "msTextSizeAdjust": "100%",
            "msoLineHeightRule": "exactly",
            "padding": 0,
          }
        }
        valign="top"
      >
        <table
          align="left"
          border={0}
          cellPadding={0}
          cellSpacing={0}
          className="mcnImageCardRightImageContentContainer"
          style={
            Object {
              "WebkitTextSizeAdjust": "100%",
              "borderCollapse": "collapse",
              "msTextSizeAdjust": "100%",
              "msoTableLspace": "0pt",
              "msoTableRspace": "0pt",
            }
          }
          width={200}
        >
          <tbody>
            <tr>
              <td
                align="center"
                className="mcnImageCardRightImageContent"
                style={
                  Object {
                    "WebkitTextSizeAdjust": "100%",
                    "msTextSizeAdjust": "100%",
                    "msoLineHeightRule": "exactly",
                    "paddingBottom": "18px",
                    "paddingLeft": "18px",
                    "paddingRight": 0,
                    "paddingTop": "18px",
                  }
                }
                valign="top"
              >
                <a
                  data-testid="sponsorHrefTest"
                  style={
                    Object {
                      "WebkitTextSizeAdjust": "100%",
                      "msTextSizeAdjust": "100%",
                      "msoLineHeightRule": "exactly",
                    }
                  }
                  target="_blank"
                >
                  <img
                    alt=""
                    className="mcnImage"
                    data-testid="sponsorSrcTest"
                    style={
                      Object {
                        "border": 0,
                        "borderRadius": "0%",
                        "height": "auto",
                        "maxWidth": "150px",
                        "msInterpolationMode": "bicubic",
                        "outline": "none",
                        "textDecoration": "none",
                        "verticalAlign": "bottom",
                      }
                    }
                    width={150}
                  />
                </a>
              </td>
            </tr>
          </tbody>
        </table>
        <table
          align="right"
          border={0}
          cellPadding={0}
          cellSpacing={0}
          className="mcnImageCardRightTextContentContainer"
          style={
            Object {
              "WebkitTextSizeAdjust": "100%",
              "borderCollapse": "collapse",
              "msTextSizeAdjust": "100%",
              "msoTableLspace": "0pt",
              "msoTableRspace": "0pt",
            }
          }
          width={346}
        >
          <tbody>
            <tr>
              <td
                className="mcnTextContent"
                style={
                  Object {
                    "WebkitTextSizeAdjust": "100%",
                    "color": "#f2f2f2",
                    "fontFamily": "Helvetica",
                    "fontSize": "14px",
                    "fontWeight": "normal",
                    "lineHeight": "150%",
                    "msTextSizeAdjust": "100%",
                    "msoLineHeightRule": "exactly",
                    "paddingBottom": "18px",
                    "paddingRight": "18px",
                    "paddingTop": "18px",
                    "textAlign": "center",
                    "wordBreak": "break-word",
                  }
                }
                valign="top"
              >
                <h1
                  className="mc-toc-title"
                  style={
                    Object {
                      "color": "#111111",
                      "display": "block",
                      "fontFamily": "\\"Merriweather Sans\\", \\"Helvetica Neue\\",Helvetica, Arial, sans-serif",
                      "fontSize": "26px",
                      "fontStyle": "normal",
                      "fontWeight": "bold",
                      "letterSpacing": "normal",
                      "lineHeight": "125%",
                      "margin": 0,
                      "padding": 0,
                      "textAlign": "center",
                    }
                  }
                >
                  <br />
                  <a
                    style={
                      Object {
                        "WebkitTextSizeAdjust": "100%",
                        "color": "#111111",
                        "fontWeight": "bold",
                        "msTextSizeAdjust": "100%",
                        "msoLineHeightRule": "exactly",
                        "textDecoration": "underline",
                      }
                    }
                  >
                    <strong
                      data-testid="sponsorChildrenTest"
                    />
                  </a>
                </h1>
              </td>
            </tr>
          </tbody>
        </table>
      </td>
    </tr>
  </tbody>
</table>
`;
